pkg/util/ec2/ec2.go

```diff
@@ -99,11 +99,23 @@ var instanceIDFetcher = cachedfetch.Fetcher{
 	},
 }
 
+var imdsv2InstanceIDFetcher = cachedfetch.Fetcher{
+	Name: "EC2 IMDSv2 InstanceID",
+	Attempt: func(ctx context.Context) (interface{}, error) {
+		return getMetadataItemWithMaxLength(ctx, imdsInstanceID, true)
```

Comment:
**❓ question:** ‏I don't understand in which case GetInstanceID can fail while GetIMDSv2InstanceID succeeds. From my understanding, it's the case if the returned hostname has a length > `metadata_endpoints_max_hostname_size` (255 by default). 

This is not the case in the example you gave in the PR description so I don't really understand in which case your changes are used